### DISCUSSION TITLE
Arch Linux WSL2: Setting up to compile

### DISCUSSION THREAD

**Discussion by do-kevin** [Etaash-mathamsetty/trigger-control#16] [Q&A]
Hi,

Sorry if this is more about WSL and Arch Linux than trigger control; I only know webdev, this is completely new to me. I've already had WSL2 installed and installed yuk7's Arch Linux.

Upon running this from the README,
```
sudo pacman -S sdl2 base-devel libdecor ttf-dejavu dbus
```

I get:
```
error: failed retrieving file 'mesa-22.2.1-1-x86_64.pkg.tar.zst' from mirrors.kernel.org : The requested URL returned error: 404
error: failed retrieving file 'guile-2.2.7-2-x86_64.pkg.tar.zst' from mirrors.kernel.org : The requested URL returned error: 404
error: failed retrieving file 'binutils-2.39-3-x86_64.pkg.tar.zst' from mirrors.kernel.org : The requested URL returned error: 404
warning: too many errors from mirrors.kernel.org, skipping for the remainder of this transaction
error: failed retrieving file 'gcc-12.2.0-1-x86_64.pkg.tar.zst' from mirrors.kernel.org : The requested URL returned error: 404
error: failed retrieving file 'llvm-libs-14.0.6-3-x86_64.pkg.tar.zst' from mirrors.kernel.org : The requested URL returned error: 404
warning: failed to retrieve some files
error: failed to commit transaction (failed to retrieve some files)
Errors occurred, no packages were upgraded.
```

What should I do?

**Comment by BananaWorks07** [comment]
try `sudo pacman -Syu`

**Reply by do-kevin** [reply]
After running `sudo pacman -Syu sdl2 base-devel libdecor ttf-dejavu dbus`

I encountered this next error:
```
error: linux-api-headers: signature from "Frederik Schwan <frederik.schwan@linux.com>" is unknown trust
:: File /var/cache/pacman/pkg/linux-api-headers-6.1.9-1-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] Y
error: glibc: signature from "Frederik Schwan <frederik.schwan@linux.com>" is unknown trust
:: File /var/cache/pacman/pkg/glibc-2.37-2-x86_64.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] Y
```
Whether I put `Y` or `N`, I end up with 
```
error: failed to commit transaction (invalid or corrupted package)
Errors occurred, no packages were upgraded.
```

**Reply by do-kevin** [reply]
All right, looks like running `pacman -Sy archlinux-keyring && pacman -Syu` first got them to go through. And installing the missing libnotify via `sudo pacman -S libnotify`

**Reply by BananaWorks07** [reply]
also make sure to run `make windows` because that's how you get a windows build and not a linux one if you want to stop depending on WSL, the windows variant doesn't need libnotify

**Reply by do-kevin** [reply]
I'm at this step: `yay -S mingw-w64-pkg-config mingw-w64-sdl2`. I cannot build `yay` so I can run that command. When I git clone yay, change directory into it, and do `makepkg -si` as a user other than `root` (and I added that user to sudoers), I receive 
```
Running makepkg as root is not allowed as it can cause permanent,
catastrophic damage to your system.
```

Eh, what do I do next to fix this?

**Reply by BananaWorks07** [reply]
don't run makepkg with sudo and make sure that you don't have root access
I remember specifically having issues with yay on archWSL, so I would try running makepkg and cloning the aur packages using git (which is the official way to do it)

**Reply by do-kevin** [reply]
I could not figure it out. Could you elaborate further? I logged in as an another user and tried to run only `makepkg` but no success.

**Reply by BananaWorks07** [reply]
is there any paticular reason you want to build it yourself? You can always grab a build from the CI

**Reply by BananaWorks07** [reply]
> is there any paticular reason you want to build it yourself? You can always grab a build from the CI

in fact you could look at the way the CI builds it without dependence on the AUR 
https://github.com/Etaash-mathamsetty/trigger-control/blob/main/.github/workflows/build.yml

**Reply by do-kevin** [reply]
> is there any paticular reason you want to build it yourself? You can always grab a build from the CI

Ah, well, I wanted to see if I can compile it on my own first. Then secondly...

I actually want to break own your code to see if I can understand how you did it. But I would need to learn C++ and SDL2.
I'm hoping to use your code or make something similar to it so I can implement it in a future Godot project that would use GDExtension.

Do you think your code would be enough to control the triggers in a videogame or is there a lot more technical stuff I don't know about that's needed to do that kind of feature?

**Reply by BananaWorks07** [reply]
> Do you think your code would be enough to control the triggers in a videogame or is there a lot more technical stuff I don't know about that's needed to do that kind of feature?

Should work, in fact you can directly just copy paste it (if godot uses SDL), my code is under MIT license

> I actually want to break own your code to see if I can understand how you did it. But I would need to learn C++ and SDL2.
I'm hoping to use your code or make something similar to it so I can implement it in a future Godot project that would use GDExtension.

go ahead, thanks to SDL the code is relatively simple, back when I used HIDAPI, it was much tougher to get bluetooth working, but once I found code for bluetooth reports (in the linux kernel for example), I was able to recreate it here, when I discovered SDL could do it by itself (no HIDAPI), I switched over to that (you can look in commit history to see how the original HIDAPI implementation worked)
